src/wallet/coincontrol.h

```diff
@@ -76,8 +77,24 @@ class CCoinControl
         vOutpoints.assign(setSelected.begin(), setSelected.end());
     }
 
+    void AddKnownCoins(const CInputCoin& coin)
+    {
+        knownCoins.insert(std::make_pair(coin.outpoint, coin));
+    }
+
+    boost::optional<CInputCoin> FindKnownCoin(const COutPoint& outpoint) const
+    {
+        boost::optional<CInputCoin> foundCoin;
+        auto it = knownCoins.find(outpoint);
+        if (it != knownCoins.end())
+            foundCoin = it->second;
+        return foundCoin;
+    }
+
 private:
     std::set<COutPoint> setSelected;
+    //! A map of known UTXO
+    std::map<COutPoint, CInputCoin> knownCoins;
```

Comment:
I briefly tried, but it changes lot's of stuff in the QT Code. I prefer doing it in a separate PR, as I fear the changes will be too hard to review.

The design of this PR will not be greatly impacted once I switch setSelected to use CInputCoin everywhere. Basically only this block https://github.com/bitcoin/bitcoin/pull/10202/files#diff-b2bb174788c7409b671c46ccc86034bdR2247  will need to be removed.